src/main/java/org/dita/dost/module/CopyFlagModule.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * This file is part of the DITA Open Toolkit project.
+ *
+ * Copyright 2010 IBM Corporation
+ *
+ * See the accompanying LICENSE file for applicable license.
+
+ */
+package org.dita.dost.module;
+
+import static org.dita.dost.util.Constants.ANT_INVOKER_EXT_PARAM_DITAOUTPUTDIR;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.util.Collection;
+
+import org.apache.tools.ant.BuildException;
+import org.apache.tools.ant.util.FileUtils;
+import org.dita.dost.exception.DITAOTException;
+import org.dita.dost.pipeline.AbstractPipelineInput;
+import org.dita.dost.pipeline.AbstractPipelineOutput;
+import org.dita.dost.util.Job.FileInfo;
+import org.dita.dost.util.URLUtils;
+
+/**
+ * Copy DITAVAL flag module.
+ */
+final class CopyFlagModule extends AbstractPipelineModuleImpl {
+
+    @Override
+    public AbstractPipelineOutput execute(final AbstractPipelineInput input)
+            throws DITAOTException {
+        final Collection<FileInfo> fis = job.getFileInfo(f -> f.isFlagImage);
+        if (!fis.isEmpty()) {
+        	try {
+        		final FileUtils fileUtils = FileUtils.newFileUtils();
+        		final File outputDirFile = new File(input.getAttribute(ANT_INVOKER_EXT_PARAM_DITAOUTPUTDIR));
+        		if (!outputDirFile.exists()) {
+        			outputDirFile.mkdir();
+        		}
+        		
+        		final URI outputDirURI = URLUtils.toURI(outputDirFile);
+        		for (final FileInfo f: fis) {
+        			final File srcFile = new File(f.src);
+        			final File destFile = new File(outputDirURI.resolve(f.uri));
```

Comment:
We should not use the temporary file name to generate the final output file. `FileInfo` has `result`, an absolute URI for the intended output file.

Reply:
I feel like I'm missing something -- `result` looks like it matches `src`, so it's an absolute URI for the source file to be generated, but not absolute for the output file, right? Basing this on what I see for both DITA topics and (non-flag) images in 3.3.2 output for HTML5. For example, with an image referenced in the source:
`<file src="file:/C:/dcs/test/jira226/simple/content/anotherpuff.png" uri="anotherpuff.png" path="anotherpuff.png" result="file:/C:/dcs/test/jira226/simple/content/anotherpuff.png" format="image">`

The output image from this build is actually 
`file:/C:/dcs/test/jira226/simple/content/output332/anotherpuff.png`

Reply:
(To be clear: I can see how using the temporary file name is wrong, if any other scheme is used we get bad output / incorrect image file names. Just not sure how to translate `result` into the correct output path.)